### PR TITLE
[SPARK-15402] [ML] [PySpark] PySpark ml.evaluation should support save/load

### PR DESCRIPTION
**Pull request by yanboliang**
## What changes were proposed in this pull request?

Since `ml.evaluation` has supported save/load at Scala side, supporting it at Python side is very straightforward and easy.
## How was this patch tested?

Add python doctest.

### CODE REVIEW

**Comment by yanboliang** [comment]
cc @jkbradley @mengxr

File: python/pyspark/ml/evaluation.py [hunk 1]
```diff
@@ -316,17 +335,27 @@ def setParams(self, predictionCol="prediction", labelCol="label",
 
 if __name__ == "__main__":
     import doctest
+    import tempfile
+    import pyspark.ml.evaluation
     from pyspark.context import SparkContext
     from pyspark.sql import SQLContext
-    globs = globals().copy()
+    globs = pyspark.ml.evaluation.__dict__.copy()
```
Comment (thunterdb) [inline python/pyspark/ml/evaluation.py]: can you explain why you need this change (and add a comment about it)?

Comment (yanboliang) [inline python/pyspark/ml/evaluation.py]: @thunterdb If we did not make this change, we can work well when using ```./bin/pyspark``` to run all tests, but it failed when run with ```./python/run-tests --modules=pyspark-ml```.This issue is weird, I still not figure out the root cause, and we did the same changes when we add other save/load.

**Comment by holdenk** [comment]
This seems pretty reasonable, do you have the time to update it to master and then maybe we could get @MLnick to take a quick look?

**Comment by yanboliang** [comment]
Resolved merge conflicts, ping @holdenk @MLnick @srowen @jkbradley to take a look when you available. This is just add Python API and should be very straight forward to move ahead. Thanks.

File: python/pyspark/ml/evaluation.py [hunk 2]
```diff
@@ -21,7 +21,8 @@
 from pyspark.ml.wrapper import JavaParams
 from pyspark.ml.param import Param, Params, TypeConverters
 from pyspark.ml.param.shared import HasLabelCol, HasPredictionCol, HasRawPredictionCol
-from pyspark.ml.common import inherit_doc
+from pyspark.ml.util import JavaMLReadable, JavaMLWritable
+from pyspark.mllib.common import inherit_doc
```
Comment (BryanCutler) [inline python/pyspark/ml/evaluation.py]: Just wondering why change to import from `mllib.common`?

**Comment by BryanCutler** [comment]
Just one question, otherwise LGTM

File: python/pyspark/ml/evaluation.py [hunk 3]
```diff
@@ -311,19 +330,25 @@ def setParams(self, predictionCol="prediction", labelCol="label",
...
         .appName("ml.evaluation tests")\
         .getOrCreate()
-    sc = spark.sparkContext
-    globs['sc'] = sc
     globs['spark'] = spark
-    (failure_count, test_count) = doctest.testmod(
-        globs=globs, optionflags=doctest.ELLIPSIS)
-    spark.stop()
-    if failure_count:
-        exit(-1)
```
Comment (BryanCutler) [inline python/pyspark/ml/evaluation.py]: should there still be an `exit(-1)` when failures?  If not then you can remove the returned variables from `doctest.testmod`

File: python/pyspark/ml/evaluation.py (hunk 2 above)
Comment (yanboliang) [inline python/pyspark/ml/evaluation.py]: Oops, typo.

File: python/pyspark/ml/evaluation.py (hunk 3 above)
Comment (yanboliang) [inline python/pyspark/ml/evaluation.py]: Thanks for remind, added check ```failure_count```.

**Comment by holdenk** [comment]
ping @MLnick & @davies for final review if they have bandwidth around this?

**Comment by yanboliang** [comment]
Since this is very straight forward, I'll merge it into master. Any comments are still appreciated, and I can address them in follow up work. Thanks for review! @holdenk @BryanCutler

**Pull request closed without merging**